crates/ty_server/src/server/api.rs

```diff
@@ -49,6 +49,15 @@ pub(super) fn request(req: server::Request) -> Task {
         >(
             req, BackgroundSchedule::LatencySensitive
         ),
+        lsp_types::request::Shutdown::METHOD => {
```

Comment:
I ended up rewriting the shutdown handling during my investigation and I sort of like this more.

Reply:
Thank you! I like this a lot, I was wondering myself whether the custom `Connection` struct could be removed with your recent changes around cancellation / retry.